sympy/series/limits.py

```diff
@@ -146,15 +146,14 @@ def doit(self, **hints):
 
         if e.is_Mul:
             if abs(z0) is S.Infinity:
-                # XXX todo: this should probably be stated in the
-                # negative -- i.e. to exclude expressions that should
-                # not be handled this way but I'm not sure what that
-                # condition is; when ok is True it means that the leading
-                # term approach is going to succeed (hopefully)
+                ztemp = Symbol("z",real = True)
```

Comment:
The logarithm is not restricted to real arguments. It is also defined for non-zero complex arguments. Even in your code `log(m).subs(z,ztemp)).is_polynomial(ztemp)` the logarithm is taken before ztemp is substituted. Besides, if `log(m).subs(z, ztemp)` a polynomial in ztemp, it is already a polynomial in z before substitution. So the substitution does not give new information.